### PR TITLE
test: add NATS restart integration test

### LINKED ISSUE #527: Add integration test: Hermes recovers after real NATS restart
All reconnect tests in test_publisher_reconnect.py mock nats.connect. There is no integration-level test that starts a real NATS server, connects Hermes, kills the server, restarts it, and verifies /health returns 200 within nats_reconnect_interval seconds. This would catch edge cases in the actual nats-py connection lifecycle (e.g. drain() behavior on a closed socket) that mocks cannot surface.

_Follow-up from #346_

### PR DESCRIPTION
**Pull request by mvillmow**
## Summary
Add integration test that verifies Hermes recovers after a real NATS server restart, catching edge cases in the actual nats-py connection lifecycle that mocks cannot surface.

## Changes
- Add test_nats_restart_integration.py with real NATS server lifecycle test (start, connect, kill, restart, verify /health)
- Update publisher.py to support integration test scenario
- Update test_publisher.py and test_publisher_callbacks.py with test infrastructure changes

## Testing
- Integration test: real NATS server restart → verify /health returns 200 within nats_reconnect_interval
- Existing reconnect mock tests in test_publisher_reconnect.py remain unchanged

## Closes
Closes #527

Generated by Claude Code via ProjectHephaestus automation.

### CODE REVIEW

File: coverage.xml [hunk 1]
```diff
@@ -0,0 +1,855 @@
+<?xml version="1.0" ?>
```
Comment (mvillmow) [inline coverage.xml]: This 855-line generated coverage artifact should not be committed (P2/YAGNI). It is not in .gitignore, so it was picked up accidentally. Remove it from the PR and add `coverage.xml` to .gitignore so it is not re-committed. Nothing in the feature depends on it.

File: tests/test_nats_restart_integration.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    env var is honored by connect()'s own get_settings() call (publisher.py:107).
+    """
+    monkeypatch.setenv("NATS_RECONNECT_INTERVAL", str(_RECONNECT_INTERVAL))
+    monkeypatch.setenv("NATS_RECONNECT_MAX_INTERVAL", str(_RECONNECT_INTERVAL))
+    monkeypatch.setenv("NATS_RECONNECT_HARD_TIMEOUT", "3.0")
+    monkeypatch.setenv("NATS_CONNECT_TIMEOUT", "3.0")
+
+
+async def _get_health_status(pub: Publisher) -> int:
+    from hermes.server import app
```
Comment (mvillmow) [inline tests/test_nats_restart_integration.py:159]: Wrong type annotation: `tmp_path` is a `pathlib.Path`, not `pytest.TempPathFactory` (that type belongs to the session-scoped `tmp_path_factory` fixture). The body relies on `tmp_path / "jsstore"`, which only works on a Path. Annotate as `tmp_path: Path` (import `from pathlib import Path`).

File: tests/test_nats_restart_integration.py [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+    raise AssertionError(f"timed out after {timeout}s waiting for: {msg}")
+
+
+def _flags_supported() -> bool:
+    """Return True only if nats-server accepts the JetStream/store/port flags we use."""
+    if _NATS_BIN is None:
+        return False
+    try:
+        out = subprocess.run([_NATS_BIN, "--help"], capture_output=True, text=True, timeout=10)
+    except (OSError, subprocess.SubprocessError):
```
Comment (mvillmow) [inline tests/test_nats_restart_integration.py:77]: `predicate: object` forces the `# type: ignore[operator]` on the call site. Annotate as `predicate: Callable[[], bool]` (from `collections.abc import Callable`) to express the contract and drop the ignore.

**Review by mvillmow — COMMENTED** [review]
Test + is_closed production fix are correct/complete; blocker is committed coverage.xml (855-line generated artifact, not gitignored).
